src/nfc_reader/nfc_reader.py

```diff
@@ -1,12 +1,108 @@
-import nfc
+import json
+import os
+import logging
+import logging.config
+import time
+
+from src.nfc_reader.poll import poll
+from src.cloud_db.cloud_db import CloudDBClient
+from src.utils.paths import (NFC_READER_TO_SCREEN_PIPE, PROJECT_DIR)
+from src.utils.utils import get_device_id, TapStatus
+
+logging.config.fileConfig(f"{PROJECT_DIR}/config/logging.conf", disable_existing_loggers=False)
+logger = logging.getLogger(os.path.basename(__file__).split('.')[0])
+
+class NFCReader:
+    """Represents a NFC Reader.
+
+    Attributes:
+        debounce_time: Minimum time interval (in seconds) to consider consecutive taps as distinct.
+        screen_pipe: File path to the FIFO used for IPC with the screen module.
+        cloud_db: An instance of CloudDBClient for database interactions.
+    """
+
+    def __init__(self,
+                 debounce_time: int = DEBOUNCE_TIME,
+                 screen_pipe: str = NFC_READER_TO_SCREEN_PIPE) -> None:
+        """
+        Initializes the NFC Reader with specified debounce time and pipe path.
+
+        Args:
+            debounce_time: An integer specifying the debounce time in seconds.
+            screen_pipe: File path to the screen FIFO.
+        """
+        self.debounce_time = debounce_time
+        self.screen_pipe = screen_pipe
+        self.cloud_db = CloudDBClient()
+        self.last_tap = 0.0
+
+
+    def handle_tap(self, tag_id) -> bool:
+        """Handles a tap event.
+
+        The method records the tap event, validates it based on the debounce time,
+        looks up the tag_id in the cloud database and then sends the result the screen FIFO.
+
+        Returns:
+            A boolean indicating whether the tap was valid (True) or not (False).
+        """
+        timestamp = time.time()
+        tap_status = TapStatus.BAD
+
+        if (timestamp - self.last_tap) >= self.debounce_time:
+            tap_status = TapStatus.GOOD
+
+        logger.info(
+            f"NFC Id: {self.device_id}, \
+              Timestamp: {timestamp}, \
+              Is Valid: {tap_status == TapStatus.GOOD}"
+        )
+
+        is_valid_tap = tap_status == TapStatus.GOOD
+        if is_valid_tap:
+            result = self.cloud_db.lookup_tag(tag_id)
```

Comment:
The result is likely a dictionary representing the NFC object. We may want to represent it as a class similar to the `Tap` class. This would apply more to the cloud_db.